### DISCUSSION TITLE
Why does `SchemaDefinition` mandate `{ RootOperationTypeDefinition+ }`?

### DISCUSSION THREAD

**Discussion by mjy9088** [graphql/graphql-spec#1131] [Q&A] [answered]
What if someone wants to define a schema with only directives or a description, without `RootOperationTypeDefinition`s?

**Comment by martinbonnin** [comment]
@mjy9088 what would be the use case of doing so? A schema without any operation isn't really useful.

**Reply by mjy9088** [reply]
I haven't used TypeSystemExtension yet, so I'm not entirely sure if such a case exists, but I was thinking that perhaps the main schema could only provide descriptions and directives, while the extended schema could provide the RootOperationTypeDefinition.

Above all, as I'm creating a parser myself rather than focusing on usage examples, I feel there's some inconsistency in the fact that only SchemaDefinition among TypeSystemDefinitionOrExtension enforces the `{ ... }` block.

**Reply by martinbonnin** [reply]
> only SchemaDefinition among TypeSystemDefinitionOrExtension enforces the { ... } block.

Don't objects and interfaces require a `{ ... }` block as well? They are required to contain fields.

**Reply by mjy9088** [reply]
None of the ones you mentioned require fields in the specifications I have seen, if I interpreted these correctly.

- https://spec.graphql.org/October2021/#ObjectTypeDefinition
- https://spec.graphql.org/October2021/#ObjectTypeExtension
- https://spec.graphql.org/October2021/#InterfaceTypeDefinition
- https://spec.graphql.org/October2021/#InterfaceTypeExtension

**Reply by martinbonnin** [reply]
@mjy9088 lists are non-empty in the GraphQL spec. See https://spec.graphql.org/October2021/#sel-FARLJNCAACCrDgpe

```
A subscript suffix “list” is shorthand for a list of one or more of that symbol, represented as an additional recursive production.
```

**Reply by mjy9088** [reply]
Yes, at least one FieldDefinition is required in FieldsDefinition.

But the FieldsDefinition seems optional (with `lookahead != {`)

**Reply by martinbonnin** [reply]
Ah, yes indeed! TIL

**Comment by benjie** [comment, marked as answer]
In general we start with the tightest/strictest definitions and then open them up as the need arises. Since there has been no need for this to be opened up, we have not done so.

**Reply by mjy9088** [reply]
Thank you, the intention is clearly understood!
